### PR TITLE
Adding List type to checkUpdate validator

### PR DESCRIPTION
**Pull request by zvikorn**
Signed-off-by: Tzvika Avni <tavni@redhat.com>

### CODE REVIEW

**Comment by zvikorn** [comment]
@vkareh PTAL

**Comment by vkareh** [comment]
I'm getting this:
```
pkg/language/checks.go:427:8: no new variables on left side of :=
make[1]: *** [Makefile:34: cmds] Error 2
```

**Comment by zvikorn** [comment]
Weird. I ran the command you gave me and got no errors. Will look for this

**Comment by vkareh** [comment]
that was fixed now, but it gives me this:
```
E: Method 'AccountsMgmt.V1.Account.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.GenericLabel.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Organization.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.ResourceQuota.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Role.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.RoleBinding.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Subscription.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.AddOn.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Cluster.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Flavour.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Ingress.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Ingresses.Update' should have exactly one struct parameter but it has 0
E: Method 'ClustersMgmt.V1.Label.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Syncset.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Account.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.GenericLabel.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Organization.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.ResourceQuota.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Role.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.RoleBinding.Update' should have exactly one list parameter but it has 0
E: Method 'AccountsMgmt.V1.Subscription.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.AddOn.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Cluster.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Flavour.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Ingress.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Ingresses.Update' should have exactly one struct parameter but it has 0
E: Method 'ClustersMgmt.V1.Label.Update' should have exactly one list parameter but it has 0
E: Method 'ClustersMgmt.V1.Syncset.Update' should have exactly one list parameter but it has 0
E: Can't read model: there were 28 errors
make: *** [Makefile:54: generate] Error 1
```

File: pkg/language/checks.go [hunk 1]
```diff
@@ -391,24 +391,40 @@ func (r *Reader) checkPost(method *concepts.Method) {
...
 
+	// Exactly one list parameter:
+	var lists []*concepts.Parameter
+	for _, parameter := range method.Parameters() {
+		if parameter.Type().IsList() {
+			lists = append(lists, parameter)
+		}
+	}
+	count := len(lists)
+	if count != 1 {
```
Comment (cben) [inline pkg/language/checks.go:413]: did you mean `count > 1` and "at most one list parameter" ?

Comment (zvikorn) [reply pkg/language/checks.go:413]: I copied the same code from func (r *Reader) checkList(method *concepts.Method) {

Comment (tzvatot) [reply pkg/language/checks.go:413]: count > 1 is the right check on PATCH list.

**Comment by zvikorn** [comment]
@vkareh good to merge?

**Comment by vkareh** [comment]
I'm still getting this error when I build the SDK with https://github.com/openshift-online/ocm-api-metamodel/pull/92 and https://github.com/openshift-online/ocm-api-model/pull/215
```
E: Method 'ClustersMgmt.V1.Ingresses.Update' should have exactly one struct parameter but it has 0
E: Method 'ClustersMgmt.V1.Ingresses.Update' should have exactly one struct parameter but it has 0
E: Can't read model: there were 2 errors
make: *** [Makefile:54: generate] Error 1
```

**Comment by vkareh** [comment]
@zvikorn - https://github.com/openshift-online/ocm-api-metamodel/pull/93

**Pull request closed without merging**